client/my-sites/checkout/checkout-thank-you/domain-transfer-to-any-user/index.tsx

```diff
@@ -16,31 +16,40 @@ interface DomainTransferToAnyUserContainerProps {
 const DomainTransferToAnyUser: React.FC< DomainTransferToAnyUserContainerProps > = ( {
 	domain,
 } ) => {
+	const upsellProps = {
+		title: translate( 'Professional email' ),
+		description: translate(
+			'85% of people trust an email address with a custom domain name over a generic one.'
+		),
+		icon: emailImage,
+		action: (
+			<Button
+				href={ getEmailManagementPath( domain, domain ) }
+				onClick={ () =>
+					recordTracksEvent( 'calypso_domain_transfer_thank_you_professional_email_click' )
+				}
+			>
+				{ translate( 'Add email' ) }
+			</Button>
+		),
+	};
+
+	const products = [ <ThankYouDomainProduct domainName={ domain } key="transferred-domain" /> ];
```

Comment:
I have concern regarding this array.

Actually, you can provide simple component, w/o array, since you expect `products: ReactNode`.
`ReactNode` includes all possible types that can be rendered in a React application: strings, numbers, ReactElements, ARRAYS, etc.

But, instead of just removing Array wrapper, I would suggest to do tiny refactoring: [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/thank-you-v2/index.tsx#L13), replace:
`products?: React.ReactNode;` with `products?: React.ReactElement | React.ReactElement[];`. In. this case we will expect array of components or one component, BUT we will be more specific. ReactNode is an extra common type, as you can see in our case - previously, we made some kind of mistake and expected one element as `ReactNode` instead of `ReactNode[]`, but it was allowed to provide arrays of components.

Did you get what I mean? Hope I haven't confused you 😅 Otherwise, I will elaborate 🙂

Reply:
Ah, I didn't realize `React.ReactElement` is more specific. Now I've learned :)

Reply:
But alsom, now, we can remove array wrapper here

const products = <ThankYouDomainProduct domainName={ domain } />;